### PR TITLE
fix: chrome detection in macOS Catalina

### PR DESCRIPTION
**Pull request by clarkdo**
Fix https://github.com/nuxt/tib/issues/29

### CODE REVIEW

**Comment by pimlie** [comment]
Thanks, did @rchl 's implementation in #30 not work out?

**Comment by rchl** [comment]
Just updated mine as I've noticed that it was matching too loosely (also apps within app [1]). That said, I'm fine with whatever works, but I think with this change I'm getting some duplicates:

```
> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -dump  | grep -E -i -o '/.*(google chrome( canary)?|chromium).app' | grep -v -E "Caches|TimeMachine|Temporary|/Volumes" | uniq
/Users/me/Downloads/Google Chrome.app
/Users/me/.Trash/Google Chrome.app
/Applications/Google Chrome.app
/Applications/Google Chrome Canary.app
/Applications/Google Chrome.app
/Applications/Google Chrome Canary.app
/Applications/Google Chrome.app
/Applications/Google Chrome Canary.app
/Users/me/workspace/nuxt/vue-meta/node_modules/puppeteer/.local-chromium/mac-674921/chrome-mac/Chromium.app
```

[1] For example: /Applications/Google Chrome.app/Contents/Frameworks/Google Chrome Framework.framework/Versions/78.0.3904.70/Helpers/Google Chrome Helper (GPU).app

**Comment by rchl** [comment]
With my fix I'm currently getting this:

```
> /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -dump  | grep -E -i '(google chrome( canary)?|chromium)\.app(\s|$)'
path:               /Users/me/.Trash/Google Chrome.app (0x2790)
path:               /Applications/Google Chrome.app (0x27a4)
path:               /Applications/Google Chrome Canary.app (0x2864)
path:               /Users/me/workspace/nuxt/vue-meta/node_modules/puppeteer/.local-chromium/mac-674921/chrome-mac/Chromium.app (0xd08)
path:               /Volumes/Google Chrome/Google Chrome.app (0xd90)
```
which is then further parsed by javascript.

File: src/utils/detectors/chrome.js [hunk 1]
```diff
@@ -45,9 +45,10 @@ export default class ChromeDetector {
       installations.push(customChromePath)
     }
     execSync(
-      `${LSREGISTER} -dump` +
-        ' | grep -E -i \'(google chrome( canary)?|chromium).app$\'' +
-        ' | awk \'{$1=""; print $0}\''
+      `${LSREGISTER} -dump |
+        grep -E -i -o '/.*(google chrome( canary)?|chromium).app' |
```
Comment (rchl) [inline src/utils/detectors/chrome.js]:
```suggestion
        grep -E -i -o '/.*(google chrome( canary)?|chromium).app(\\s|$)' |
```

File: src/utils/detectors/chrome.js [hunk 2]
```diff
@@ -45,9 +45,10 @@ export default class ChromeDetector {
...
+        grep -v -E "Caches|TimeMachine|Temporary|/Volumes" |
+        uniq`
```
Comment (rchl) [inline src/utils/detectors/chrome.js]:
```suggestion
        sort | uniq`
```

File: src/utils/detectors/chrome.js [hunk 3]
```diff
@@ -45,9 +45,10 @@ export default class ChromeDetector {
       installations.push(customChromePath)
     }
     execSync(
-      `${LSREGISTER} -dump` +
-        ' | grep -E -i \'(google chrome( canary)?|chromium).app$\'' +
-        ' | awk \'{$1=""; print $0}\''
+      `${LSREGISTER} -dump |
+        grep -E -i -o '/.*(google chrome( canary)?|chromium).app' |
+        grep -v -E "Caches|TimeMachine|Temporary|/Volumes" |
```
Comment (rchl) [inline src/utils/detectors/chrome.js]:
```suggestion
        grep -v -E "Caches|TimeMachine|Temporary|/Volumes|\.Trash" |
```

**Comment by clarkdo** [comment]
I didn’t notice there is a pr already,  close this.

@rchl Have you verified your pr in other version macOS ?

**Comment by rchl** [comment]
@clarkdo It still runs on CI so I'd say that is good enough verification.

Since you've decided to close yours, I can apply your filtering out of unwanted results to my PR.

**Pull request closed without merging**